### ISSUE TITLE
Is this app using AWS?

### ISSUE THREAD

**Issue by 2lu3** [tyage/slack-patron#136]
Hi!

I deployed this app locally.
However, some channels can't complete loading like this.
![image](https://user-images.githubusercontent.com/16276366/197544975-5b84e1d4-c873-4ff4-a4fb-c3f7ac5ebc31.png)

It seems that there is a problem because I'm not using aws.
If it's correct, it looks like there is no information to use aws for this app.
Could you give me some hints (not documentation, because it costs your time) to use aws for this app?.

```
container_name  | 2022-10-24 14:01:05 - Aws::Sigv4::Errors::MissingCredentialsError - unable to sign request without credentials set:
container_name  |      /usr/local/bundle/gems/aws-sigv4-1.5.1/lib/aws-sigv4/signer.rb:707:in `fetch_credentials'
container_name  |      /usr/local/bundle/gems/aws-sigv4-1.5.1/lib/aws-sigv4/signer.rb:406:in `presign_url'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/presigner.rb:254:in `block in sign_but_dont_send'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/seahorse/client/plugins/request_callback.rb:85:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/streaming_retry.rb:71:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/s3_signer.rb:132:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/s3_host_id.rb:17:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/xml/error_handler.rb:10:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/transfer_encoding.rb:26:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/helpful_socket_errors.rb:12:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/s3_signer.rb:110:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/redirects.rb:20:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/retry_errors.rb:360:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/http_checksum.rb:19:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/endpoint_pattern.rb:30:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/accelerate.rb:67:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/checksum_algorithm.rb:136:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/bucket_dns.rb:35:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/dualstack.rb:41:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/expect_100_continue.rb:22:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/bucket_name_restrictions.rb:26:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/arn.rb:62:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/rest/handler.rb:10:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/recursion_detection.rb:18:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/user_agent.rb:13:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/endpoint_discovery.rb:80:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/seahorse/client/plugins/endpoint.rb:47:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/param_validator.rb:26:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/arn.rb:88:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/seahorse/client/plugins/raise_response_errors.rb:16:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/skip_whole_multipart_get_checksums.rb:18:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/sse_cpk.rb:24:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/dualstack.rb:27:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/plugins/accelerate.rb:56:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/checksum_algorithm.rb:111:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:16:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/idempotency_token.rb:19:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/param_converter.rb:26:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/seahorse/client/plugins/request_callback.rb:71:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/aws-sdk-core/plugins/response_paging.rb:12:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/seahorse/client/plugins/response_target.rb:24:in `call'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/presigner.rb:194:in `block in handle_presigned_url_context'
container_name  |      /usr/local/bundle/gems/aws-sdk-core-3.131.5/lib/seahorse/client/request.rb:72:in `send_request'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/presigner.rb:146:in `_presigned_request'
container_name  |      /usr/local/bundle/gems/aws-sdk-s3-1.114.0/lib/aws-sdk-s3/presigner.rb:78:in `presigned_url'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:31:in `sign_file'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:45:in `block (2 levels) in sign_message_files'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:44:in `each'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:44:in `block in sign_message_files'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:42:in `each'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:42:in `sign_message_files'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:111:in `messages'
container_name  |      /usr/local/slack-patron/viewer/viewer.rb:199:in `block in <top (required)>'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1686:in `call'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1686:in `block in compile!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1023:in `block (3 levels) in route!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1042:in `route_eval'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1023:in `block (2 levels) in route!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1071:in `block in process_route'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1069:in `catch'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1069:in `process_route'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1021:in `block in route!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1018:in `each'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1018:in `route!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1140:in `block in dispatch!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1112:in `block in invoke'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1112:in `catch'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1112:in `invoke'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1135:in `dispatch!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:949:in `block in call!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1112:in `block in invoke'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1112:in `catch'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1112:in `invoke'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:949:in `call!'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:938:in `call'
container_name  |      /usr/local/bundle/gems/rack-protection-2.2.2/lib/rack/protection/xss_header.rb:18:in `call'
container_name  |      /usr/local/bundle/gems/rack-protection-2.2.2/lib/rack/protection/path_traversal.rb:16:in `call'
container_name  |      /usr/local/bundle/gems/rack-protection-2.2.2/lib/rack/protection/json_csrf.rb:26:in `call'
container_name  |      /usr/local/bundle/gems/rack-protection-2.2.2/lib/rack/protection/base.rb:50:in `call'
container_name  |      /usr/local/bundle/gems/rack-protection-2.2.2/lib/rack/protection/base.rb:50:in `call'
container_name  |      /usr/local/bundle/gems/rack-protection-2.2.2/lib/rack/protection/frame_options.rb:31:in `call'
container_name  |      /usr/local/bundle/gems/rack-2.2.4/lib/rack/logger.rb:17:in `call'
container_name  |      /usr/local/bundle/gems/rack-2.2.4/lib/rack/common_logger.rb:38:in `call'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:255:in `call'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:248:in `call'
container_name  |      /usr/local/bundle/gems/rack-2.2.4/lib/rack/head.rb:12:in `call'
container_name  |      /usr/local/bundle/gems/rack-2.2.4/lib/rack/method_override.rb:24:in `call'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:218:in `call'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1553:in `block in call'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1769:in `synchronize'
container_name  |      /usr/local/bundle/gems/sinatra-2.2.2/lib/sinatra/base.rb:1553:in `call'
container_name  |      /usr/local/bundle/gems/puma-5.6.4/lib/puma/configuration.rb:252:in `call'
container_name  |      /usr/local/bundle/gems/puma-5.6.4/lib/puma/request.rb:77:in `block in handle_request'
container_name  |      /usr/local/bundle/gems/puma-5.6.4/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
container_name  |      /usr/local/bundle/gems/puma-5.6.4/lib/puma/request.rb:76:in `handle_request'
container_name  |      /usr/local/bundle/gems/puma-5.6.4/lib/puma/server.rb:441:in `process_client'
container_name  |      /usr/local/bundle/gems/puma-5.6.4/lib/puma/thread_pool.rb:147:in `block in spawn_thread'
```

**Comment by 2lu3** [comment]
I guess AWS is used to store files posted to slack.
However, I don’t know which AWS service is used.
Also, I don’t know the permission scope of aws credentials.

**Comment by tyage** [comment]
Hi, I guess this is a known bug and already fixed in Slack Patron v3.0.1.
https://github.com/tyage/slack-patron/pull/133

Can you update to the latest version?

**Comment by 2lu3** [comment]
Sorry about it.
When I update the docker image, it works.
Thank you!
